### PR TITLE
2137 select star syntax

### PR DESCRIPTION
**Pull request by laabidiHend**
#2137 Fixes

### CODE REVIEW

**Review by fm-117 — CHANGES_REQUESTED** [review]

File: TypeCobol/Compiler/Sql/Model/FullSelect.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿namespace TypeCobol.Compiler.Sql.Model
+{
+    public class FullSelect : SqlObject
```
[the remark this answers is not in the document]
Comment (fm-117) [reply TypeCobol/Compiler/Sql/Model/FullSelect.cs:3]: See https://www.ibm.com/docs/en/db2-for-zos/12?topic=queries-fullselect, fullselect supports all the options of a SELECT operation and most importantly allows to combine multiple subselects using set operators (UNION, EXCEPT, INTERSECT)

File: TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs [hunk 2]
```diff
@@ -1,6 +1,11 @@
-﻿using TypeCobol.Compiler.CodeElements;
+﻿using TypeCobol.Compiler.AntlrUtils;
+using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.Parser;
 using TypeCobol.Compiler.Parser.Generated;
+using TypeCobol.Compiler.Scanner;
 using TypeCobol.Compiler.Sql.CodeElements.Statements;
+using TypeCobol.Compiler.Sql.Model;
+using TypeCobol.Compiler.Sql.Nodes;
```
[the remark this answers is not in the document]
Comment (fm-117) [reply TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs:8]: Yes I've checked with ReSharper, you should remove:

```csharp
using TypeCobol.Compiler.Parser;
using TypeCobol.Compiler.Scanner;
using TypeCobol.Compiler.Sql.Nodes;
```

File: TypeCobol/Compiler/Sql/CodeElements/Statements/SelectStatement.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.Parser.Generated;
+using TypeCobol.Compiler.Sql.Model;
```
[the remark this answers is not in the document]
Comment (fm-117) [reply TypeCobol/Compiler/Sql/CodeElements/Statements/SelectStatement.cs:8]: Checked myself, everything can be removed except `using TypeCobol.Compiler.CodeElements;` and `using TypeCobol.Compiler.Sql.Model;`

File: TypeCobol/Compiler/Sql/Nodes/Select.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.Nodes;
+using TypeCobol.Compiler.Sql.CodeElements.Statements;
+
+namespace TypeCobol.Compiler.Sql.Nodes
+{ /// <summary>
```
Comment (fm-117) [inline TypeCobol/Compiler/Sql/Nodes/Select.cs:6]: Indent is wrong, summary block should start on its own line

**Review by fm-117 — CHANGES_REQUESTED** [review]

**Review by fm-117 — APPROVED** [review]
Ok for me unless we decide to improve the Token capture mechanism.

File: TypeCobol.Test/Parser/Programs/Cobol85/ExecSql/ExecSqlWithSelectStar.rdzMix.txt [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿       IDENTIFICATION DIVISION.
+       PROGRAM-ID. DVZZMFT3.
+Line 3[8,21] <27, Error, Syntax> - Syntax error : extraneous input 'DATADIVISION.' expecting {ExecStatementText, ExecStatementEnd, CommitStatement, SelectStatement}
```
Comment (smedilol) [inline TypeCobol.Test/Parser/Programs/Cobol85/ExecSql/ExecSqlWithSelectStar.rdzMix.txt:3]: Error on `data division` ?

File: TypeCobol/Compiler/Sql/Model/SelectClause.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    public abstract class Selection
+    {
+        public abstract SelectionType Type { get; }
+    }
+
+    public class StarSelection : Selection
+    {
+        public override SelectionType Type => SelectionType.Star;
```
[the remark this answers is not in the document]
Comment (smedilol) [reply TypeCobol/Compiler/Sql/Model/SelectClause.cs:29]: I don't know what is the best solution.
We need to discuss about `ConsumedTokens` and also properties like `SymbolInformationForTokens`, `StorageAreaDefinitions`, `StorageAreaReads` ...

**Review by smedilol — CHANGES_REQUESTED** [review]

**Review by mayanje — APPROVED** [review]

**Review by smedilol — APPROVED** [review]

**Pull request merged**